cmd/podman/info.go

```diff
@@ -88,6 +89,9 @@ func infoCmd(c *cliconfig.InfoValues) error {
 
 	var out formats.Writer
 	infoOutputFormat := c.Format
+	if strings.Join(strings.Fields(infoOutputFormat), "") == "{{json.}}" {
```

Comment:
This looks strange, why do you need this `strings` processesing?  
Whis is 
```
if c.Format == "{{json.}}" {
```
Not enough?

Reply:
For the fix I used code from an approved merge found in PR #2688 which solved an nearly identical problem.
If that looks off to you then the other piece of code should change as well.

Reply:
i have verified that c.Format == "{{json.}}" does not work.
./bin/podman info --format '{{ json .}}'
Error: template parsing error: template: image:1: function "json" not defined

This fix is more-or-less a band-aid.  Would it be worth investigating to determine the larger issue?

Reply:
@baude Why was this necessary?  Is this just some because the c.Format is a StirngVarP?  Versus a StringVar?

Reply:
This is special casing to support a strange Docker convention where `--format '{{ json. }}'` is the same as `--format json`.

Reply:
The reason for `strings.Fields` is whitespace removal.